File: studies/2022.06_rayleigh_noise_update/investigate_fft_routine/compare_fft.cpp
// C/C++ Includes
#include <iostream>
#include <cstdlib>
#include <sstream>
#include <fstream>
#include <stdio.h>
#include <vector>
#include <cmath>
#include <algorithm>

//AraRoot Includes
#include "RawAtriStationEvent.h"
#include "UsefulAtriStationEvent.h"
#include "FFTtools.h"

//ROOT Includes
#include "TTree.h"
#include "TFile.h"
#include "TGraph.h"
#include "TCanvas.h"

#include "helper.h"

RawAtriStationEvent *rawAtriEvPtr;

using namespace std;

int main(int argc, char **argv)
{

    TFile *fpIn = new TFile("/disk20/users/brian/ARA/event_003315.root");
    fpIn->cd();
    TTree *eventTree = (TTree*) fpIn->Get("eventTree");
    eventTree->SetBranchAddress("event", &rawAtriEvPtr);
    int numEntries = eventTree->GetEntries();
    std::cout<<"Num Entries "<<numEntries<<std::endl;

    for(int event=0; event<numEntries; event++){ //loop over those entries
        eventTree->GetEntry(event); //get the event
                
        if(rawAtriEvPtr->isSoftwareTrigger()){
            std::cout<<"On event "<<event<<std::endl;

            UsefulAtriStationEvent *realAtriEvPtr = new UsefulAtriStationEvent(rawAtriEvPtr, AraCalType::kLatestCalib);
            TGraph *grRaw = realAtriEvPtr->getGraphFromRFChan(0);
            TGraph *grInt = FFTtools::getInterpolatedGraph(grRaw, 0.5);
            TGraph *grPad = FFTtools::padWaveToLength(grInt, 1024);

            TGraph *grSpectrum = do_with_FFTtools(grPad);
                
            delete grSpectrum, grPad, grInt, grRaw;
            delete realAtriEvPtr;
        }

    }
    fpIn->Close();
    delete fpIn;
}

File: studies/2022.06_rayleigh_noise_update/investigate_fft_routine/compare_fft.py
import ROOT
from ROOT import gInterpreter, gSystem
import numpy as np
import matplotlib.pyplot as plt
import os 
import helper as helper

ROOT.gSystem.Load(os.environ.get('ARA_UTIL_INSTALL_DIR')+"/lib/libAraEvent.so")
ROOT.gSystem.Load(os.environ.get('ARA_DEPS_INSTALL_DIR')+"/lib/libRootFftwWrapper.so")
gInterpreter.ProcessLine('#include "' + os.environ.get('ARA_DEPS_INSTALL_DIR') + '/include/FFTtools.h"')
# gInterpreter.ProcessLine('#include "/cvmfs/ara.opensciencegrid.org/trunk/centos7/source/ARA_cvmfs/build/include/FFTtools.h"')


file = ROOT.TFile.Open("/disk20/users/brian/ARA/event_003315.root")
eventTree = file.Get("eventTree")
rawEvent = ROOT.RawAtriStationEvent()
eventTree.SetBranchAddress("event", ROOT.AddressOf(rawEvent))

num_events = eventTree.GetEntries()
print("total events {}".format(num_events))

for event in range(num_events):

    eventTree.GetEntry(event)
    if(rawEvent.isSoftwareTrigger()):
        print("We have a software trigger on event {}".format(event))

        usefulEvent = ROOT.UsefulAtriStationEvent(rawEvent, ROOT.AraCalType.kLatestCalib)
        grRaw = usefulEvent.getGraphFromRFChan(0)
        grInt = ROOT.FFTtools.getInterpolatedGraph(grRaw, 0.5)
        grPad = ROOT.FFTtools.padWaveToLength(grInt, 1024)
        
        helper.do_fft_with_python(grPad)


        del grPad, grInt, grRaw, usefulEvent



        # c = ROOT.TCanvas()
        # c.cd()
        # grRaw.Draw("ALP")
        # grPad.Draw("Lsame")
        # grPad.SetLineColor(ROOT.kRed)
        # c.Print("root_waveform_{}.png".format(event))
